js/cryptocom.js

```diff
@@ -0,0 +1,1026 @@
+                        'private/create-order',
+                        'private/cancel-order',
+                        'private/cancel-all-orders',
+                        'private/get-order-history',
+                        'private/get-open-orders',
+                        'private/get-order-detail',
+                        'private/get-trades',
+                    ],
+                },
+            },
+            'fees': {
+                'trading': {
+                    'maker': 0.004,
+                    'taker': 0.004,
+                },
+            },
+            'options': {},
+            // https://exchange-docs.crypto.com/spot/index.html#response-and-reason-codes
+            'exceptions': {
+                'exact': {
+                    '10001': ExchangeError,
+                    '10002': PermissionDenied,
+                    '10003': PermissionDenied,
+                    '10004': BadRequest,
+                    '10005': PermissionDenied,
+                    '10006': DDoSProtection,
+                    '10007': InvalidNonce,
+                    '10008': BadRequest,
+                    '10009': BadRequest,
+                    '20001': BadRequest,
+                    '20002': InsufficientFunds,
+                    '30003': BadSymbol,
+                    '30004': BadRequest,
+                    '30005': BadRequest,
+                    '30006': BadRequest,
+                    '30007': BadRequest,
+                    '30008': BadRequest,
+                    '30009': BadRequest,
+                    '30010': BadRequest,
+                    '30013': BadRequest,
+                    '30014': BadRequest,
+                    '30016': BadRequest,
+                    '30017': BadRequest,
+                    '30023': BadRequest,
+                    '30024': BadRequest,
+                    '30025': BadRequest,
+                    '40001': BadRequest,
+                    '40002': BadRequest,
+                    '40003': BadRequest,
+                    '40004': BadRequest,
+                    '40005': BadRequest,
+                    '40006': BadRequest,
+                    '40007': BadRequest,
+                    '50001': BadRequest,
+                },
+            },
+        });
+    }
+
+    nonce () {
+        return this.milliseconds ();
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'] + '/' + path;
+        const query = this.omit (params, this.extractParams (path));
+        if (api === 'public') {
+            if (Object.keys (query).length) {
+                url += '?' + this.urlencode (query);
+            }
+        } else {
+            this.checkRequiredCredentials ();
+            const id = 1;
+            const nonce = this.nonce ().toString ();
+            const requestParams = this.extend ({}, params);
+            const paramsKeys = Object.keys (requestParams).sort ();
```

Comment:
it seems the `.sort` cause issues in Python. Anyone can help me?

Reply:
@Johann-S check how `this.keysort()` is used in the other exchanges for this purpose.

Reply:
it works well! Thank you 👍